bevy_lint/src/groups.rs

```diff
@@ -0,0 +1,66 @@
+use crate::LintGroup;
+use rustc_lint::Level;
+
+/// A group of deny-by-default lints that check for outright wrong or useless code.
+///
+/// These lints are carefully picked to be free of false-positives. You should avoid
+/// `#[allow(...)]`-ing these lints without a _very_ good reason.
+pub static CORRECTNESS: LintGroup = LintGroup {
+    name: "bevy::correctness",
+    level: Level::Deny,
+};
+
+/// A group similar to [`CORRECTNESS`] that checks for suspicious or usually wrong code.
+///
+/// As compared to [`CORRECTNESS`], it may be possible that the linted code may be written
+/// intentionally. Even still, you usually want to fix these lints instead of `#[allow(...)]`-ing
+/// them.
```

Comment:
```suggestion
/// The linted code may have been written intentionally, but should probably still be fixed.
```
Perhaps? Slightly clearer IMHO, YMMV.

Reply:
Yup! These lint group descriptions may need a bit more fixing-up after this gets merged.